substrate/frame/staking/src/lib.rs

```diff
@@ -1235,6 +1235,16 @@ pub trait DisablingStrategy<T: Config> {
 		slash_era: EraIndex,
 		currently_disabled: &Vec<u32>,
 	) -> Option<u32>;
+
+	/// Returns the maximum number of validators that can be disabled at a given point in time.
+	///
+	/// This method is needed only for the implementation of `VersionUncheckedMigrateV14ToV15` in
+	/// staking pallet. Feel free to remove after this migration is no longer needed because in
+	/// theory `DisablingStrategy` can be anything. It even (in theory!) can have no disabling
+	/// threshold. However we need to know the maximum number of disabled validators expected by the
+	/// `DisablingStrategy` implementation so that we can migrate `OffendingValidators` without
+	/// breaking any invariants.
+	fn disable_threshold(validators_len: usize) -> usize;
```

Comment:
Name is not ideal. `max_disabled_validators` or something would be more descriptive. Also if the disabling strategy does not need to have that value we should return an `Option` here. Then this is generally useful and we don't need to break a public interface by later removal.

Reply:
I'm using the strategy implementation directly in the migration. This way there is no need to expose the disabling limit. What do you think?